### PR TITLE
node: Handle "upgrade" event

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: node/http.ts [hunk 1]
```diff
@@ -462,7 +475,18 @@ class ServerImpl extends EventEmitter {
...
           const res = new ServerResponse(reqEvent);
-          this.emit("request", req, res);
+          if (req.upgrade && this.listenerCount("upgrade") > 0) {
+            Deno.upgradeHttp(req._webRequest).then(([conn, head]) => {
+              const socket = new Socket({
+                handle: new TCP(constants.SERVER, conn),
+              });
+              this.emit("upgrade", req, socket, new Buffer(head));
+            }).catch((err) => {
+              console.error("upgrade errored", err);
```
Comment (bartlomieju) [inline node/http.ts:485]: This should probably emit `close` or `error` event

**Comment by kt3k** [comment]
@bartlomieju Is it possible to write test case for upgrade behavior?

**Comment by bartlomieju** [comment]
Yes, I will write a test (or enable some Node tests) before landing.

**Comment by bartlomieju** [comment]
@kt3k I would like to land this PR without tests - I verified that it works with both Vite and express. Adding tests requires larger refactor that I'd like to tackle next week or two weeks after.

**Pull request merged**